src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/LinkDesignerBase.cs

```diff
@@ -21,6 +21,15 @@ public virtual void LogMessage (string message)
 			Context.LogMessage (message);
 		}
 
+		public virtual void LogError (int code, string error)
+		{
+#if ILLINK
+			Context.LogMessage (MessageContainer.CreateCustomErrorMessage (error, code, origin: new MessageOrigin ()));
+#else   // !ILLINK
+			Console.Error.WriteLine ($"error XA{code}: {error}");
```

Comment:
If `ILLINK` is not defined, we should be within `Xamarin.Android.Build.Tasks.dll`, right?  In which case we should have a "real" MSBuild `Task` that can raise "real" MSBuild error messages, instead of going through `Console.Error.WriteLine()`?

Reply:
This is done in https://github.com/xamarin/xamarin-android/pull/8066/files#diff-6bfa8cb4f52234d33d427a9aa52682179ea4e86d986fa7b2c6c26504ffcd6d96R145 by overriding this method. The Console.Error.WriteLine here is just a backup.